fence/sync/sync_users.py

```
@@ -328,42 +332,94 @@ def _parse_yaml(self, filepath, encrypted=True):
         user_project = dict()
         user_info = dict()
 
-        if filepath is None:
-            return user_project, user_info
-
+        data = dict()
         try:
             with self._read_file(filepath, encrypted=encrypted) as stream:
                 data = yaml.safe_load(stream)
-                users = data.get('users', {})
-                for username, details in users.iteritems():
-                    privileges = {}
-
-                    try:
-                        for project in details.get('projects', {}):
-                            privileges[project['auth_id']] = set(
-                                project['privilege']
-                            )
-                    except KeyError as e:
-                        self.logger.info(e)
-                        continue
+        except IOError as e:
+            self.logger.error(e)
+            return dict(), dict()
 
-                    user_info[username] = {
-                        'email': details.get('email', username),
-                        'display_name': details.get('display_name', ''),
-                        'phone_number': details.get('phone_number', ''),
-                        'tags': details.get('tags', {}),
-                        'admin': details.get('admin', False),
-                    }
+        users = data.get('users', {})
+        for username, details in users.iteritems():
+            # users should occur only once each; skip if already processed
+            if username in user_project:
+                self.logger.error(
+                    'user `{}` occurs multiple times (skipping)'
+                    .format(username)
+                )
```

didn't see it being skipped? I think we should just raise the exception here too.

---

"skipping" as in `continue`s to the next user. I'll change it to an `EnvironmentError`.

---

changed in 0cee0f2